templates/modelExpt/run_modelComp_audapter.m

```diff
@@ -0,0 +1,217 @@
+            adjustment = adjustments{setting2change};
+
+            expt = adjust_experimentSettings(expt, h_fig, adjustment);
+        end
+
+        % plot trial number in experimenter view
+        cla(h_sub(1))
+        ctrltxt = sprintf('trial: %d/%d, cond: %s',itrial,expt.ntrials,expt.listConds{itrial});
+        h_trialn = text(h_sub(1),0,0.5,ctrltxt,'Color','black', 'FontSize',30, 'HorizontalAlignment','center'); %#ok<NASGU> 
+
+        % set text
+        txt2display = expt.listWords{itrial};
+        color2display = expt.colorvals{expt.allColors(itrial)};
+
+        % set new perturbation
+        p.pertAmp = expt.shiftMags(itrial) * ones(1, 257);
+        p.pertPhi = expt.shiftAngles(itrial) * ones(1, 257);
+        Audapter('setParam','pertAmp',p.pertAmp)
+        Audapter('setParam','pertPhi',p.pertPhi)
+
+        % run trial in Audapter
+        Audapter('reset'); %reset Audapter
+        fprintf('starting trial %d\n',itrial)
+        Audapter('start'); %start trial
+        fprintf('Audapter started for trial %d\n',itrial)
+
+        % display stimulus
+        h_text(1) = draw_exptText(h_fig,.5,.5,txt2display, 'Color',color2display, 'FontSize',stimtxtsize, 'HorizontalAlignment','center');
+        pause(expt.timing.stimdur);
+
+        % stop trial in Audapter and get data
+        Audapter('stop');
+        fprintf('Audapter ended for trial %d\n',itrial)
+        data = AudapterIO('getData');
+
+        % plot shifted spectrogram
+        subplot_expt_spectrogram(data, p, h_fig, h_sub);
+
+        % check if participant spoke loud enough for amplitude to cross required threshold
+        bGoodTrial = check_rmsThresh(data, expt.amplcalc, h_sub(3));
+        subplot(h_sub(3))
+        yyaxis right
+        hline(2, 'r', '-'); % reference line at 2, where the vowel starts
+
+        if ~bGoodTrial %first check if amplitude crossed threshold
+            h_text(2) = draw_exptText(h_fig,.5,.2,'Please speak a little louder','FontSize',40,'HorizontalAlignment','center','Color','y');
+
+            % pause for viewing duration feedback
+            pause(expt.timing.visualfbdur);
+
+            % clear screen
+            delete_exptText(h_fig,h_text)
+            clear h_text
+        elseif expt.bDurFB(itrial) %plot duration feedback, if needed for this trial
+            % plot duration feedback
+            [h_dur,success] = plot_duration_feedback(h_fig(stim), data, expt.durcalc); % original line
+            if isfield(expt,'success')
+                expt.success(itrial) = success;
+            end
+
+            CloneFig(h_fig(stim),h_fig(dup))
+
+            % pause for viewing duration feedback
+            pause(expt.timing.visualfbdur);
+
+            % clear screen
+            delete_exptText(h_fig,[h_text h_dur])
+            clear h_text h_dur
+        else
+            % clear screen
+            delete_exptText(h_fig,h_text)
+            clear h_text
+        end
```

Comment:
Probably could do something clever here to move this:
```
            % clear screen
            delete_exptText(h_fig,h_text)
            clear h_text
```
outside the loop. Maybe by clearing h_dur inside the loop in the case where it exists, and clearing the rest outside it? Not a high-priority change, but it may make this easier to understand, since we're basically doing the same thing in all three cases (clearing the screen).

Reply:
agreed it could be better. we'll leave it for now.